inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -1751,6 +1751,7 @@ protected List<BeanDefinitionReference> resolveBeanDefinitionReferences() {
             final SoftServiceLoader<BeanDefinitionReference> definitions = SoftServiceLoader.load(BeanDefinitionReference.class, classLoader);
             beanDefinitionReferences = new ArrayList<>(300);
             definitions.collectAll(beanDefinitionReferences, BeanDefinitionReference::isPresent);
+            beanDefinitionReferences.removeIf(Objects::isNull);
```

Comment:
seems like this should happen in SoftServiceLoader instead

Reply:
I'm not exactly sure where the `null` came from... not directly from the `SoftServiceLoader`, but from one of the `ServiceCollector`/`RecursiveActionValuesCollector`s. But at the moment, I'm not able to reproduce the issue that lead me to make this change... certainly not in local dev.  I'll need to check again on WebSphere and see if it's only there.

Reply:
I was unable to reproduce the behavior that prompted this change. And I was still able to deploy and run both a Micronaut and Grails application using the other changes from this patch, so I have reverted this change.